src/Auth.php

```diff
@@ -289,7 +289,7 @@ public function connection_ldap($ldap_method, $login, $password, ?bool &$error =
             return false;
         }
 
-        $this->ldap_connection   = AuthLDAP::tryToConnectToServer($ldap_method, $login, $password);
+        $this->ldap_connection = AuthLDAP::tryToConnectToServer($ldap_method, $ldap_method['rootdn'], (new GLPIKey())->decrypt($ldap_method['rootdn_passwd']));
```

Comment:
I missed something during my latest review.

If you pass $login and $password params, it will first try to connect using the rootdn/rootdn_password, and then, if it fails, try to connect using the given $login/$password. So the connection using the rootdn/rootdn_password will be used for all sub-sequent operations, unless it fails. Changing this piece of code will just remove this fallback that is probably here to permit to configure a LDAP without having to know admin credentials.

I guess the problem is that, since we bind the connection to the user account (in `$bind_result = $this->user_found && @ldap_bind($this->ldap_connection, $dn, $password);`), then the operations made in `User::getFromLDAP()` are not using the root account anymore.

I do not know what is the best solution to fix this. Binding back the root account depends on many parameters, so I am not sure how it should be done. For instance, on some configurations, the user binding is omited as the authentication relies on a certificate (see `AuthLDAP::connectToServer()`). Creating a new connection just to check this bind operation result can also be a problem, as it would generate extra connections.

Reply:
So your first commit was probably a better solution.